From: =?UTF-8?q?Mika=20Pfl=C3=BCger?= <debian@mikapflueger.de>
Date: Sun, 4 Mar 2012 03:22:56 +0100
Subject: xen policy adjustments: * Allow xenconsoled_t capability
 sys_tty_config and create unix_dgram_socket * Label xenfs_t and
 allow xend etc to use it * Allow noatsecure for Xen domains so that
 LD_PRELOAD will work across a domain transition. Also dontaudit
 searching of the sysadm home dir and allow xend_t to manage
 xenstored_var_run_t * Use the Lenny paths for xm, xend, xenstored,
 and xenconsoled. Add some extra permissions that Xen needs * Make
 lvm support optional

---
 policy/modules/system/xen.te |   28 +++++++++++++++++++++++++---
 1 files changed, 25 insertions(+), 3 deletions(-)

diff --git a/policy/modules/system/xen.te b/policy/modules/system/xen.te
index c4d18e8..0fff178 100644
--- a/policy/modules/system/xen.te
+++ b/policy/modules/system/xen.te
@@ -83,6 +83,9 @@ type xend_var_lib_t;
 files_type(xend_var_lib_t)
 # for mounting an NFS store
 files_mountpoint(xend_var_lib_t)
+fs_getattr_xattr_fs(xend_t)
+# for /var/lib/python-support/python2.5/.path
+files_read_var_lib_files(xend_t)
 
 # log files
 type xend_var_log_t;
@@ -320,7 +323,9 @@ locallogin_dontaudit_use_fds(xend_t)
 
 logging_send_syslog_msg(xend_t)
 
-lvm_domtrans(xend_t)
+optional_policy(`
+	lvm_domtrans(xend_t)
+')
 
 miscfiles_read_localization(xend_t)
 miscfiles_read_hwdata(xend_t)
@@ -341,6 +346,13 @@ xen_stream_connect_xenstore(xend_t)
 
 netutils_domtrans(xend_t)
 
+unconfined_dontaudit_search_home_dirs({ xend_t xenconsoled_t xenstored_t })
+ifdef(`distro_debian', `
+# xend uses LD_PRELOAD or similar for libxenctrl.so
+allow xend_t { xenconsoled_t xenstored_t }:process noatsecure;
+')
+allow xend_t xenstored_var_run_t:file manage_file_perms;
+
 optional_policy(`
 	brctl_domtrans(xend_t)
 ')
@@ -354,12 +366,16 @@ optional_policy(`
 # Xen console local policy
 #
 
-allow xenconsoled_t self:capability { dac_override fsetid ipc_lock };
+allow xenconsoled_t self:capability { dac_override fsetid ipc_lock sys_tty_config };
 allow xenconsoled_t self:process setrlimit;
 allow xenconsoled_t self:unix_stream_socket create_stream_socket_perms;
 allow xenconsoled_t self:fifo_file rw_fifo_file_perms;
+allow xenconsoled_t self:unix_dgram_socket create_socket_perms;
+
+# for /usr/lib/pt_chown
+libs_exec_lib_files(xenconsoled_t)
 
-allow xenconsoled_t xen_devpts_t:chr_file rw_term_perms;
+allow xenconsoled_t xen_devpts_t:chr_file { setattr rw_term_perms };
 
 # pid file
 manage_files_pattern(xenconsoled_t, xenconsoled_var_run_t, xenconsoled_var_run_t)
@@ -377,6 +393,7 @@ dev_rw_sysfs(xenconsoled_t)
 domain_dontaudit_ptrace_all_domains(xenconsoled_t)
 
 files_read_etc_files(xenconsoled_t)
+corecmd_search_bin(xenconsoled_t)
 files_read_usr_files(xenconsoled_t)
 
 fs_list_tmpfs(xenconsoled_t)
@@ -428,6 +445,10 @@ manage_dirs_pattern(xenstored_t, xenstored_var_lib_t, xenstored_var_lib_t)
 manage_files_pattern(xenstored_t, xenstored_var_lib_t, xenstored_var_lib_t)
 manage_sock_files_pattern(xenstored_t, xenstored_var_lib_t, xenstored_var_lib_t)
 files_var_lib_filetrans(xenstored_t, xenstored_var_lib_t, { file dir sock_file })
+allow xend_t xenstored_var_lib_t:dir rw_dir_perms;
+allow xend_t xenstored_var_lib_t:file unlink;
+corecmd_search_bin(xenstored_t)
+fs_manage_xenfs_dirs(xenstored_t)
 
 stream_connect_pattern(xenstored_t, evtchnd_var_run_t, evtchnd_var_run_t, evtchnd_t)
 
@@ -472,6 +493,7 @@ manage_files_pattern(xm_t, xend_var_lib_t, xend_var_lib_t)
 manage_fifo_files_pattern(xm_t, xend_var_lib_t, xend_var_lib_t)
 manage_sock_files_pattern(xm_t, xend_var_lib_t, xend_var_lib_t)
 files_search_var_lib(xm_t)
+files_read_kernel_img(xm_t)
 
 allow xm_t xen_image_t:dir rw_dir_perms;
 allow xm_t xen_image_t:file read_file_perms;
